File: api/Bookshop.Application/Features/Customers/Queries/Authenticate/AuthenticateResponse.cs
﻿using Bookshop.Application.Features.Response;

namespace Bookshop.Application.Features.Customers.Queries.Authenticate
{
    public class AuthenticateResponse : BaseResponse
    {
        public AuthenticateResponse() : base()
        {

        }
        public CustomerResponseDto? Customer { get; set; }
        public string? Token { get; set; }
    }
}

File: ui/src/app/dto/handler-response/customer/authenticate/authenticate-response.ts
import { CustomerResponseDto } from "../../../customer/customer-response-dto";
import { BaseResponse } from "../../../response/base-response";

export class AuthenticateResponse extends BaseResponse {

    customer: CustomerResponseDto;
    token: string;

    constructor(message?: string, error?: boolean) {
        super(message, error);
        this.customer = new CustomerResponseDto();
        this.token = '';
    }
}